php/elFinderVolumeGoogleDrive.class.php

```diff
@@ -0,0 +1,1549 @@
+            	$refresh_token = $client->fetchAccessTokenWithRefreshToken();
+            	$client->setAccessToken($refresh_token);
+           	$this->options['accessToken'] = $refresh_token;
+            	$this->service = new Google_Service_Drive($client);
+            }
+        } catch (Exception $e) {
+            return $e->getMessage();
+        }
+        return true;
+    }
+    
+    /**
+     * Get dat(googledrive metadata) from GoogleDrive
+     * 
+     * @param string $path
+     * @return array googledrive metadata
+     */
+    private function getPathtoName($path)
+    {
+        $itemId = basename($path);
+        $opts = [
+                'fields' => self::FETCHFIELDS_GET
+            ];
+            
+        $res = $this->service->files->get($itemId, $opts);
+        return $res['name'];
+    }
+    
+    /**
+     * Get dat(googledrive metadata) from GoogleDrive
+     * 
+     * @param string $path
+     * @return array googledrive metadata
+     */
+    private function getDBdat($path)
+    {
+        if ($path == $this->root) {
+            $this->root == '/' ? $itemId = 'root' : $itemId = basename($this->root);
+			$opts = [
+				'fields' => self::FETCHFIELDS_GET
+			];            
+			return $this->service->files->get($itemId, $opts);
+        }
+        
+        empty($this->HasdirsCache[$path]) ? $HasPath = $path : $HasPath = $this->HasdirsCache[$path][0];
+        $itemId = basename($HasPath);
+
+		try{
+			$opts = [
+				'fields' => self::FETCHFIELDS_GET
+				];
+			return $this->service->files->get($itemId, $opts);
+		} catch (Exception $e) {
+			return array();
+		}
+    }
+    
+
+    /*********************************************************************/
+    /*                               FS API                              */
+    /*********************************************************************/
+
+    /**
+     * Close opened connection
+     *
+     * @return void
+     * @author Dmitry (dio) Levashov
+     **/
+    public function umount()
+    {
+    }
+        
+    /**
+     * Parse line from googledrive metadata output and return file stat (array)
+     *
+     * @param  string  $raw  line from ftp_rawlist() output
+     * @return array
+     * @author Dmitry Levashov
+     **/
+    protected function parseRaw($raw)
+    {
+        $stat = array();
+
+        $stat['rev']        = isset($raw['id']) ? $raw['id'] : 'root';
+        $stat['name']        = isset($raw['name']) ? $raw['name'] : '';
+        $stat['mime']        = $raw['mimeType'] == self::DIRMIME ? 'directory' :
+                              ($raw['mimeType'] == 'image/bmp' ? 'image/x-ms-bmp' :  $raw['mimeType']);
+        $stat['size']        = $raw['mimeType'] == self::DIRMIME ? 0 : (int)$raw['size'];
+        $stat['ts']            = isset($raw['modifiedTime']) ? strtotime($raw['modifiedTime']) : $_SERVER['REQUEST_TIME'];
+        $stat['dirs']        = $raw['mimeType'] == self::DIRMIME ? 1 : 0;
```

Comment:
This is not good idea. Please remove this line, And to make detect `dirs` correctly.